REFERENCE.md

```diff
@@ -0,0 +1,240 @@
+	'domain'         => 'localhost'
+}
+# XXX: The consumer secret should be in a properly secured credential store, and *NEVER* checked into version control.
+# Set values in application.rb.
+consumer_secret = Rails.configuration.consumer_secret
+
+data_request = { 'limit' => 50 }
+
+init = Learnosity::Sdk::Request::Init.new(
+        'data',
+        security_packet,
+        consumer_secret,
+        data_request,
+	'get'
+)
+
+request = init.generate
+
+Net::HTTP.post_form URI('https://data.learnosity.com/v1/itembank/items'), request
+```
+
+### Recursive Queries
+
+tl;dr: not currently implemented
+
+Some requests are paginated to the `limit` passed in the request, or some
+server-side default. Responses to those requests contain a `next` parameter in
+their `meta` property, which can be placed in the next request to access another
+page of data.
+
+For the time being, you can iterate through pages by looping over the
+`Init#new`/`Init#generate`/`Net::HTTP#post_form`, updating the `next` attribute
+in the request.
+
+```ruby
+response = JSON.parse(res.body)
+if ( !response['meta']['next'].nil? \
+		and !response['meta']['records'].nil? and response['meta']['records'] > 0)
+	data_request['next'] = response['meta']['next']
+end
+```
+
+This will `require 'json'` to be able to parse the response.
+
+See `examples/simple/init_data.rb` for an example.
+
+### Generating UUIDs
+
+You will need to generate UUIDs. You can use the Ruby `securerandom`
+module for this purpose.
+
+```ruby
+require 'securerandom'
+
+p SecureRandom.uuid
+```
+
+### Rails
+
+You can bootstrap a Ruby-on-Rails project using this SDK by doing the
+following.
+
+Note, this is a quick 0 to 100 in 10s examples of how to use this
+SDK with Rails. As such, it takes many unwise shortcuts in the architecture of
+the app; do not take this as the example of a good Rails app.
+
+First, generate a skeleton project
+
+    rails new lrn-sdk-rails
+    cd lrn-sdk-rails
+
+Add the `learnosity-sdk` as a dependency to this project
+
+    echo "gem 'learnosity-sdk' >> Gemfile
+    bundle install
+
+Create a default controller
+
+    rails generate controller Index index
+
+Add the `require` for the SDK at the top of the newly created controller,
+`app/controllers/index_controller.rb`, and insert the assessment configuration
+there (taken from [items-api-demo], and truncated for legibility). Add the demo 
+consumer key and secret to [application.rb].
+
+```ruby
+require 'learnosity/sdk/request/init'
+
+class IndexController < ApplicationController
+  @@security_packet = {
+    # XXX: This is a Learnosity Demos consumer; replace it with your own consumer key. Set values in application.rb.
+    'consumer_key'   => Rails.configuration.consumer_key,
+    'domain'         => 'localhost'
+  }
+# XXX: The consumer secret should be in a properly secured credential store, and *NEVER* checked into version control.
+# Set values in application.rb.
+consumer_secret = Rails.configuration.consumer_secret
```

Comment:
Might be worth to point back to the README.md, about the `Rails.configuration` bit, as we don't define an `application.rb` in here, but assume it exists.